plugins/optimization-detective/optimization.php

```diff
@@ -168,11 +206,10 @@ function od_optimize_template_output_buffer( string $buffer ): string {
 	do_action( 'od_register_tag_visitors', $tag_visitor_registry );
 
 	$link_collection      = new OD_Link_Collection();
-	$processor            = new OD_HTML_Tag_Processor( $buffer );
 	$tag_visitor_context  = new OD_Tag_Visitor_Context( $processor, $group_collection, $link_collection );
 	$current_tag_bookmark = 'optimization_detective_current_tag';
 	$visitors             = iterator_to_array( $tag_visitor_registry );
-	while ( $processor->next_open_tag() ) {
+	do {
```

Comment:
is this any different, or just prefer do/while over while?

Reply:
It's because I needed to check above if the first tag is an open `HTML` tag. So when we get down here we don't need to call `next_open_tag()` since we're already at the first open tag.

Plus, I can probably count on my hand the number of times I've been able to use a `do`/`while` loop!